QueryDispatch/Listeners/PatientEventListener.cs

```diff
@@ -61,29 +78,26 @@ protected override Task ExecuteAsync(CancellationToken stoppingToken)
                         }
                         catch (ConsumeException e)
                         {
-                            _logger.LogError($"Consume failure, potentially schema related: {e.Error.Reason}");
-                            var potentialFacilityId = Encoding.UTF8.GetString(e.ConsumerRecord.Message.Key);
-
-                            var auditValue = new AuditEventMessage
+                            var facilityId = Encoding.UTF8.GetString(e.ConsumerRecord.Message.Key);
+                            var converted_record = new ConsumeResult<string, string>()
                             {
-                                FacilityId = potentialFacilityId,
-                                Action = AuditEventType.Query,
-                                ServiceName = "QueryDispatch",
-                                EventDate = DateTime.UtcNow,
-                                Notes = $"Kafka PatientEvent consume failure, potentially schema related \nException Message: {e.Error}",
+                                Message = new Message<string, string>()
+                                {
+                                    Key = facilityId,
+                                    Value = Encoding.UTF8.GetString(e.ConsumerRecord.Message.Value),
+                                    Headers = e.ConsumerRecord.Message.Headers
+                                }
                             };
 
-                            ProduceAuditEvent(auditValue, e.ConsumerRecord.Message.Headers);
+                            _consumeResultDeadLetterExceptionHandler.HandleException(converted_record, new DeadLetterException("Consume Result exception: " + e.InnerException.Message, AuditEventType.Create), facilityId);
 
-                            //TODO: We will eventually have Error/Retry workflows implemented
                             _patientEventConsumer.Commit();
```

Comment:
Should we be calling `Commit(e.ConsumerRecord)` here?  (Does that even work?)  Could this arg-less `Commit` potentially commit more offsets than we want?

Reply:
This is a good point. From what I've seen so far, you can't commit the ex.ConsumeRecord from the exception. Will dig into this a bit.

Reply:
Something like this may work out. 
`_patientEventConsumer.Commit(new List<TopicPartitionOffset> { e.ConsumerRecord.TopicPartitionOffset });`
Testing it out now.